lib/handler/connect.c

```diff
@@ -685,18 +686,23 @@ static int tcp_start_connect(struct st_connect_generator_t *self, struct st_serv
     return 1;
 }
 
-static h2o_iovec_t udp_get_next_chunk(const char *start, size_t len, size_t *to_consume, int *skip)
+static h2o_iovec_t udp_get_next_chunk(const char *start, size_t len, size_t *to_consume, int *skip, int datagram_format)
 {
     const uint8_t *bytes = (const uint8_t *)start;
     const uint8_t *end = bytes + len;
-    uint64_t chunk_type, chunk_length;
+    uint64_t chunk_type, chunk_length, capsule_context_id;
 
     chunk_type = ptls_decode_quicint(&bytes, end);
     if (chunk_type == UINT64_MAX)
         return h2o_iovec_init(NULL, 0);
     chunk_length = ptls_decode_quicint(&bytes, end);
     if (chunk_length == UINT64_MAX)
         return h2o_iovec_init(NULL, 0);
+    if (datagram_format == H2O_DATAGRAM_FORMAT_RFC) {
+        capsule_context_id = ptls_decode_quicint(&bytes, end);
+        if (capsule_context_id == UINT64_MAX)
+            return h2o_iovec_init(NULL, 0);
+    }
 
     /* chunk is incomplete */
     if (end - bytes < chunk_length)
```

Comment:
This check should be looking at the distance between `end` and what `bytes` was before the parsing of the `capsule_context_id` move `bytes` forward.  The `chunk_length` includes the length of the encoded context id because the context id is inside the capsule.  Please see doohara/h2o#1 and my comment attached to a few lines below.

Reply:
Interesting, yes that makes sense, thanks for also testing for this with Quiche. The PR looks good, i'll merge it.